src/Microsoft.OpenApi.Readers/ParseNodes/ParseNode.cs

```diff
@@ -26,8 +26,7 @@ protected ParseNode(ParsingContext parsingContext, OpenApiDiagnostic diagnostic)
 
         public MapNode CheckMapNode(string nodeName)
         {
-            var mapNode = this as MapNode;
-            if (mapNode == null)
+            if (!(this is MapNode mapNode))
             {
                 throw new OpenApiException($"{nodeName} must be a map/object");
```

Comment:
>OpenApiException [](start = 26, length = 16)

OpenApiReaderException?

Reply:
Yeah, probably should be.

---
In reply to: [194831575](https://github.com/Microsoft/OpenAPI.NET/pull/276#discussion_r194831575) [](ancestors = 194831575)